src/python/WMCore/Services/DBS/DBS3Writer.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+"""
+_DBSReader_
+
+Read/Write DBS Interface
+
+"""
+from __future__ import print_function, division
+from builtins import str
+import logging
+
+from dbs.apis.dbsClient import DbsApi
+from dbs.exceptions.dbsClientException import dbsClientException
+
+from WMCore.Services.DBS.DBSErrors import DBSWriterError, formatEx3
+from WMCore.Services.DBS.DBS3Reader import DBS3Reader
+
+
+class DBS3Writer(DBS3Reader):
+    """
+    _DBSReader_
+
+    General API for writing data to DBS
+    """
+
+    def __init__(self, url, writeUrl, logger=None, **contact):
+
+        # instantiate dbs api object
+        try:
+            super(DBS3Writer, self).__init__(url=url)
+            self.dbs = DbsApi(writeUrl, **contact)
+            self.logger = logger or logging.getLogger(self.__class__.__name__)
+        except dbsClientException as ex:
+            msg = "Error in DBSWriter with DbsApi\n"
+            msg += "%s\n" % formatEx3(ex)
+            raise DBSWriterError(msg)
+
+    def setDBSStatus(self, dataset, status):
+        """
+        The function to set the DBS status of an output dataset
+        :param dataset: Dataset name
+        :return: True if operation is successful, False otherwise
+        """
+
+        try:
+            self.dbs.updateDatasetType(dataset=dataset,
+                                     dataset_access_type=status)
```

Comment:
indentation needs fixing. I would suggest to remove the blank line at 44 as well.

Reply:
BTW, we should fetch the output of this call and log the error in case it happens; or return it to the caller module.
By parsing the output of that call, we can also get rid of of the `getDBSStatus` implementation in the lines below.

Reply:
I also thought that but the function returns `None` if the call is successful and throws an exception o/w. I would prefer it to return True or False respectively. Since it does not do that, I handled it with `getDBSStatus` calls to make it more intuitive (according to me :) ) . We can also handle it:
- Return True if `updateDatasetType` returns None
- Return False o/w

Which one do you prefer?

Reply:
Hmm, too bad it returns None in case the call is successful..

>    Return True if updateDatasetType returns None
>    Return False o/w

This is my preference. Given that - if there is an exception - the error would likely be on the server side, I would suggest to change the log level from exception to error, such that we do not pollute the service logs with unnecessary information.

Reply:
I applied your suggestions Alan, but `setDBSStatus` still returns True if the API call is successful, False otherwise. This looks quite intuitive to me. Do you agree?